### PR TITLE
chore(argo-workflows): update workflows version to most 3.0.7

### PR DESCRIPTION
**Pull request by searsaw**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
Hello 👋 ,
Thanks for you contribution. I think we should also sync these changes here:
- https://github.com/argoproj/argo-workflows/pull/5875
- https://github.com/argoproj/argo-workflows/pull/6007

These changes were introduced in 3.0.4+

**Comment by searsaw** [comment]
> Hello 👋 ,
> Thanks for you contribution. I think we should also sync these changes here:
> 
> * [argoproj/argo-workflows#5875](https://github.com/argoproj/argo-workflows/pull/5875)
> * [argoproj/argo-workflows#6007](https://github.com/argoproj/argo-workflows/pull/6007)
> 
> These changes were introduced in 3.0.4+

Very cool. I'll tackle this later tonight.

File: charts/argo-workflows/templates/controller/workflow-controller-deployment.yaml [hunk 1]
```diff
@@ -77,12 +77,16 @@ spec:
           ports:
             - name: metrics
               containerPort: {{ .Values.controller.metricsConfig.port }}
+            - containerPort: 6060
           livenessProbe:
```
Comment (vladlosev) [inline charts/argo-workflows/templates/controller/workflow-controller-deployment.yaml:81]: I think this should be configurable. In the current configuration the controller may be down up to four minutes until Kubernetes will detect that, and many people will want a faster reaction. The entire section may be put into the config, with the value moved into in `values.yaml`:
```
  livenessProbe: {{ .Values.controller.livenessProbe | toYaml | nindent 12 }}
```

Comment (searsaw) [reply charts/argo-workflows/templates/controller/workflow-controller-deployment.yaml]: good call. this has been updated

**Review by vladlosev — APPROVED** [review]
LGTM

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**